recordio/recordio_test.go

```diff
@@ -67,7 +67,7 @@ func TestReadWriteEndToEndDirectIO(t *testing.T) {
 	require.NoError(t, err)
 
 	reader := func() ReaderI {
-		reader, err := newFileReaderWithFactory(tmpFile.Name(), DirectIOFactory{})
+		reader, err := NewFileReader(ReaderPath(tmpFile.Name()))
```

Comment:
do you mind to make this buffer factory an option too? this path now doesn't run the DirectIO check anymore

Reply:
The DirectIOFactory is in the fonction. So i Think that this does exactly the same thing as previously. Sorry, I don't understand our comment.

Reply:
maybe you want a WithFactory() option ? Is it what you think ?

Reply:
I added a IoFactory Interface. Is it ok for you ?

Reply:
yes, but you need to pass the DirectIO factory here, it's otherwise not testing it

Reply:
Done